ocean_lib/models/erc20_enterprise.py

```diff
@@ -14,14 +14,14 @@ class ERC20Enterprise(ERC20Token):
     CONTRACT_NAME = "ERC20TemplateEnterprise"
 
     def buy_from_fre_and_order(
-        self, order_params: dict, fre_params: dict, from_wallet: Wallet
+        self, order_params: tuple, fre_params: tuple, from_wallet: Wallet
     ) -> str:
         return self.send_transaction(
             "buyFromFreAndOrder", (order_params, fre_params), from_wallet
         )
 
     def buy_from_dispenser_and_order(
-        self, order_params: dict, dispenser_address: str, from_wallet: Wallet
+        self, order_params: tuple, dispenser_address: str, from_wallet: Wallet
```

Comment:
In cases where we have a tuple type we should be able to accept both regular tuples, dictionaries and our own variety of NamedTuples defined in models_structures.py. Web3.py supports both tuples and dictionaries and our tests are mixed in this aspect. If the web3.py PR for named tuples is approved, I will start refactoring on all these parameters to allow this. However, as long as the tests pass I don't want to handle this right now. We should stick to the toWei changes for now.